ThunderBasics/AccessibilityRefreshingViewController.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  AccessibilityRefreshingViewController.swift
+//  ThunderBasics-iOS
+//
+//  Created by Simon Mitchell on 21/08/2019.
+//  Copyright © 2019 threesidedcube. All rights reserved.
+//
+
+import UIKit
+
+/// A UIViewController which subscribes to accessibility setting change notifications such as `darkerSystemColorsStatusDidChangeNotification` and calls an overrideable function whenever they do change.
+open class AccessibilityRefreshingViewController: UIViewController, UIContentSizeCategoryAdjusting {
+    
+    /// Observer listening to dynamic font changes
+    private var dynamicChangeObserver: NSObjectProtocol?
+    
+    /// Observer listening to accessibility setting changed notifications
+    private var accessibilityObservers: [Any] = []
+
+    public var adjustsFontForContentSizeCategory: Bool = true
+    
+    /// A list of notification names that should cause the table view to redraw itself
+    public var accessibilityRedrawNotificationNames: [Notification.Name] = [
+        UIAccessibility.darkerSystemColorsStatusDidChangeNotification,
+        UIAccessibility.boldTextStatusDidChangeNotification,
+        UIAccessibility.grayscaleStatusDidChangeNotification,
+        UIAccessibility.invertColorsStatusDidChangeNotification,
+        UIAccessibility.reduceTransparencyStatusDidChangeNotification
+    ]
+    
+    open override func viewWillAppear(_ animated: Bool) {
+        
+        super.viewWillAppear(animated)
+        
+        dynamicChangeObserver = NotificationCenter.default.addObserver(forName: UIContentSizeCategory.didChangeNotification, object: nil, queue: .main) { [weak self] (notification) in
+            guard let strongSelf = self, strongSelf.adjustsFontForContentSizeCategory else { return }
+            strongSelf.accessibilitySettingsDidChange()
+        }
+        
+        // Notification names that it makes sense to redraw on
+        let accessibilityNotifications: [Notification.Name] = [
```

Comment:
Same as the ThunderTable PR, is there a reason why there's a separate array here? :D

Reply:
Yeah, so I figured I could try and do some magic with removing and adding notification names when `accessibilityRedrawNotificationNames` is set, but then we have to be pretty careful with making sure not to double-add or double-remove observers, and we also don't have to keep check as to whether the screen has appeared or not in that `setter`.

This way we add the notifications once, remove them once, but it still obeys the user's wishes as to when to actually refresh!